### ISSUE TITLE
Website: multi-language font filtering

### ISSUE THREAD

**Issue by HugoCortell** [google/fonts#6690]
#This issue also works for requesting features. 

**Describe the issue**
I wish to search for fonts that contain character sets for not only Latin languages, but Cyrillic, Japanese and Chinese. At the moment, the website's font language filter only lets you filter for one language at the time, making it impossible to see if a font is capable of supporting all the languages I target until I download it and manually test it with each language.

**To Reproduce**
Try to filter by language, try to select two languages, notice only one gets selected.

**Expected behavior**
You can select multiple languages

**Additional context**
Currently, my workaround is to feed it text in multiple languages, but this workflow is not optimal.

**Comment by eliheuer** [comment]
Great suggestion, thank you! I agree, this would be useful. For example in Israel you often see signs with Hebrew, Arabic, and Latin all together. It would be helpful to narrow down a font search to only see fonts supporting all three scripts.

**Comment by thlinard** [comment]
Monotype Fonts allows multiple selection of languages. This is very useful in a multitude of scenarios, for example "pan-European" needs: you want to cover all European languages, you need support for Cyrillic, Greek and Latin (rather extensive).

**Comment by sujitmahapatra** [comment]
HTML Structure:

```<div id="language-filter">
  <label for="languages">Select Languages:</label>
  <select id="languages" multiple>
    <option value="latin">Latin</option>
    <option value="cyrillic">Cyrillic</option>
    <option value="japanese">Japanese</option>
    <option value="chinese">Chinese</option>
    <!-- Add more language options as needed -->
  </select>
  <button id="apply-filter">Apply Filter</button>
</div>
```

JavaScript Functionality:

```document.getElementById('apply-filter').addEventListener('click', function () {
  var selectedLanguages = Array.from(document.getElementById('languages').selectedOptions).map(option => option.value);

  // Apply the filter logic here based on selectedLanguages
  // You may need to update the font display based on the selected languages
});
```

**Comment by thlinard** [comment]
Hi @davelab6 

This will truly be a big improvement. If you had the resources to make this happen, that would be great.

**Comment by tomasdev** [comment]
It's worth noting languages are not the same as script. A language can be written in multiple scripts and it is my impression you're interested in multiple scripts rather than languages.

**Comment by thlinard** [comment]
Hi @tomasdev,

Thanks for the update!

> It's worth noting languages are not the same as script. A language can be written in multiple scripts and it is my impression you're interested in multiple scripts rather than languages.

Indeed. And right now, the "Language" filter is more a "Charset" filter. And that's not very user-friendly. I understand for example what "Latin Extended" means, but probably not everyone. And indeed, if I'm looking for Kazakh support, I might want to make sure that both Kazakh Cyrillic and Kazakh Latin are supported.

> This is an important difference, because Cyrillic script has many characters that some get used in some languages, and some don't. Uzbek, Bosnian, Old Russian, Romany... all use cyrillic and may use it differently.

Yes, that's what "Cyrillic" and "Cyrillic Extended" are for. It would probably be more user-friendly to let users search for "Uzbek" or "Serbian".

> Same applies to Chinese, traditional vs simplified, and even from region to region, which is why there are fonts for Noto Sans TC, SC, HK, etc.

Indeed. But this does not address the current feature request. The request concerns cases where one would like to ensure that the fonts one is looking for will take into account all the linguistic needs of a project. For example, if I have to communicate in several European and Central Asian languages, I don't want 1 font for French, 1 different font for Polish, 1 font for Ukrainian, 1 for Kazakh, and yet another one for Greek (which is what the filter currently invites me to do, with 5 different, mutually exclusive options: "Latin", "Latin Extended", "Cyrillic", "Cyrillic Extended", "Greek"). I want to easily find 1 font that covers all my needs, without having to cross-reference 5 different searches.

> An update that's being released provides Writing System and Language filtering. Although still not multiple, should somewhat help in some of these queries.

Yes, still not multiple, but looks promising.

> In this same update, you'll be able to search in the search box for "cyrillic japanese" and get fonts that support both.

Oh, that's great! Thanks!

**Comment by davelab6** [comment]
> right now, the "Language" filter is more a "Charset" filter. And that's not very user-friendly

Amusingly we did UXR studies on this and found that users understood the best what a "languages" label on a dropdown would open up vs other label options, and then once opened, they didn't stumble at all when presented with writing system names (even technical-implementation version like `Latin Extended`). 

So actually this is proven user-friendly ;) 

But not sophisticated, which is what we are upgrading towards, as Tom explained

**Comment by thlinard** [comment]
> > right now, the "Language" filter is more a "Charset" filter. And that's not very user-friendly
> 
> Amusingly we did UXR studies on this and found that users understood the best what a "languages" label on a dropdown would open up vs other label options, and then once opened, they didn't stumble at all when presented with writing system names (even technical-implementation version like `Latin Extended`).
> 
> So actually this is proven user-friendly ;)

Really? And without Dunning-Kruger bias? It seems to me difficult to guess, if I didn't already know, that "Latin" is enough for German, but for a language of a border country like Poland, you must choose "Latin Extended", but if you choose "Vietnamese", you actually get German + Polish + Vietnamese most of the time…

A colleague told me some time ago that she had an international project. I asked her if she would need Cyrillic: she answered that no, Russia was not in the target countries. After checking, she needed support for Bulgarian, for which she was unaware of the use of the Cyrillic alphabet. If she had had to do the research, would she have chosen "Cyrillic"?

> But not sophisticated, which is what we are upgrading towards, as Tom explained

This looks promising, indeed.

**Comment by thlinard** [comment]
Hi @tomasdev,

On second thought, without multi-selection, it might be worse than before. What exactly does the "Language" menu allow in the new system? A selection of charsets like before, or a list of actual languages, like English, French, Spanish, etc.?

**Comment by davelab6** [comment]
> A colleague told me

We tested people working with their own language. 

> without Dunning-Kruger bias?

Surely with. But we concluded people figure it out.

**Comment by thlinard** [comment]
Hmm, I see, it's different, then. But the scenario I described is not an isolated case: it's common for creative people from one country to have to create templates for a project that will then be translated into local languages. And a Hungarian colleague told me that it's recurrent that he curses against "stupid Americans" (but it also works with "stupid French" 😅) who choose fonts for international projects, projects that will then have to be adapted using fonts that don't have the glyphs for his language.

**Comment by davelab6** [comment]
Sure, its been like this since 2013 though ;)

**Comment by thlinard** [comment]
Yes, that's why I tell my colleagues: the easiest thing to do when you need to find fonts on Google Fonts (or elsewhere, Adobe Fonts is not better in this respect) that have the language support you want is… to ask me. But when I saw that Monotype Fonts (https://enterprise.monotype.com/) allowed multiple selection for languages, finally I saw some hope.

**Comment by thlinard** [comment]
Could the "UX/UI Suggestion" label be added to this request? Thanks!

**Comment by thlinard** [comment]
I think this request can now be closed.

Thank you very much, this is a great improvement to the UI of https://fonts.google.com/
